python/ray/experimental/workflow/storage/path_based.py

```diff
@@ -0,0 +1,161 @@
+from abc import abstractmethod
+
+import ray
+from typing import Any, Dict, Callable
+from ray.experimental.workflow.common import StepID
+from ray.experimental.workflow.storage.base import (
+    Storage, ArgsType, data_load_error, data_save_error)
+
+# constants used in filesystem
+OBJECTS_DIR = "objects"
+STEPS_DIR = "steps"
+STEP_INPUTS_METADATA = "inputs.json"
+STEP_OUTPUTS_METADATA = "outputs.json"
+STEP_ARGS = "args.pkl"
+STEP_OUTPUT = "output.pkl"
+STEP_FUNC_BODY = "func_body.pkl"
+CLASS_BODY = "class_body.pkl"
+WORKFLOW_META = "workflow_meta.json"
+WORKFLOW_PROGRESS = "progress.json"
+
+
+class PathBasedStorage(Storage):
```

Comment:
Should this just be renamed to `Storage`? (If you were to implement a database-backed storage, would it also subclass this?)

Reply:
I think for kv-based database, maybe yes (use path as key). But likely no for SQL-based database, because it does not have the concept of "path".

Reply:
You could have a storage table with schema as follows:

``
string path;
string key;
string value;
INDEX (path, key);
```

So I think it would also work. How about just renaming this to Storage?

Reply:
after discussing with Yi, we agree that we can just use path for all kinds of database